WordPress/src/main/java/org/wordpress/android/ui/voicetocontent/VoiceToContentDialogFragment.kt

```diff
@@ -38,11 +43,41 @@ class VoiceToContentDialogFragment : BottomSheetDialogFragment() {
     ): View = ComposeView(requireContext()).apply {
         setContent {
             AppTheme {
-                VoiceToContentScreen(viewModel)
+                VoiceToContentScreen(
+                    viewModel = viewModel,
+                    onRequestPermission = { requestAllPermissionsForRecording() },
+                    hasPermission = { hasAllPermissionsForRecording() }
+                )
             }
         }
     }
 
+    private val requestMultiplePermissionsLauncher = registerForActivityResult(
+        ActivityResultContracts.RequestMultiplePermissions()
+    ) { permissions ->
+        val areAllPermissionsGranted = permissions.entries.all { it.value }
+        if (areAllPermissionsGranted) {
+            viewModel.startRecording()
+        } else {
+            // Handle permissions denied case
+            // todo pantelis handle permissions denied case
```

Comment:
We need to review Figma for when/how we are showing permission denied.

Reply:
There is no additional information in Figma for this. I have pushed a commit with Google's recommended approach. If the user has already denied the permission and tries to tap on the mic icon, an alert dialog is shown informing them. They are then navigated to the app's settings to grant the permission if they wish.

Reply:
Perfect ... we also double check file permission before accessing the file in the network call. We can't assume that we have permission to use the file at that point. If we don't have permission, it will be returned as an error. We'll work on presenting network errors separately.